src/shared/messaging/port-rpc.ts

```diff
@@ -236,7 +236,15 @@ export class PortRPC<OnMethod extends string, CallMethod extends string>
     // send the "close" event through the message channel when the window
     // containing the sending port is unloaded.
     if (!('onclose' in MessagePort.prototype) || forceUnloadListener) {
-      this._listeners.add(currentWindow, 'unload', () => {
+      this._listeners.add(currentWindow, 'unload', event => {
+        // Ignore custom events which use the same name. This works around an
+        // issue in VitalSource.
```

Comment:
Using custom events with the name of standard events it's a bit hacky 😅 

I can only imagine how tricky this was to discover.